src/hooks/useCheckImage.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from "react";
+import { useAsyncFn } from "react-use";
+
+export const useCheckImage = (
+  src: string
+): {
+  isValid: boolean;
+  isLoading: boolean;
+  error: string | undefined;
+} => {
+  const [{ loading, error }, checkImage] = useAsyncFn(async () => {
```

Comment:
You can use ` useAsync` and get rid of the `useEffect` just like your other PR.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/2002/commits/4f7bca683dd8443d753e222cf319790bf2a047ef